packages/clarity-js/src/layout/mutation.ts

```diff
@@ -145,6 +150,42 @@ async function process(): Promise<void> {
     task.stop(timer);
 }
 
+function track(m: MutationRecord, timer: Metric): string {
+  let value = m.target ? dom.get(m.target.parentNode) : null;
+  if (value) {
+    let inactive = time() > activePeriod;
+    // We use selector, instead of id, to determine the key because in some cases
+    // repeated mutations can cause elements to be destroyed and then recreated as new DOM nodes
+    // In those cases, IDs will change however the selector (which is relative to DOM xPath) remains the same
+    let key = [value.selector, m.attributeName, m.addedNodes ? m.addedNodes.length : 0, m.removedNodes ? m.removedNodes.length : 0].join();
+    // Initialize an entry if it doesn't already exist
+    history[key] = key in history ? history[key] : [0];
+    // Lookup any pending nodes queued up for removal, and process them now if we suspended a mutation before
+    if (inactive === false && history[key][0] >= Setting.MutationSuspendThreshold) { processNodeList(history[key][1], Source.ChildListRemove, timer); }
```

Comment:
A JS question: Why prefer this use of an array over an object with two properites (one for the time and the other for hte queued nodes) where the two parts are separated. Is it a performance or memory consideration?

Reply:
We discussed it offline.